cmds/create-ffs/create-ffs.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"crypto/sha1"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/linuxboot/fiano/pkg/guid"
+	"github.com/linuxboot/fiano/pkg/uefi"
+	"github.com/linuxboot/fiano/pkg/visitors"
+)
+
+var (
+	debug      = flag.Bool("d", false, "Enable debug prints")
+	outfile    = flag.String("o", "", "output file, default is stdout")
+	name       = flag.String("name", "", "Name to include in UI section")
+	filetype   = flag.String("type", "DRIVER", "UEFI filetype")
+	version    = flag.String("version", "1.0", "File version")
+	guidString = flag.String("guid", "", "File GUID")
+	depex      = flag.String("depex", "", "Space-separated protocol guid dependencies or TRUE")
+	compress   = flag.Bool("compress", false, "Wrap section data in a compressed section")
+	auto       = flag.Bool("auto", false, "Attempt to determine section types from file extensions")
+
+	fType  uefi.FVFileType
+	fGUID  *guid.GUID
+	depOps []uefi.DepExOp
```

Comment:
Why are you using global variables for these?

Reply:
done